### PR TITLE
Release notes for 2.2.10

### PR DESCRIPTION
**Pull request by jeff-matthews**
## Purpose of this pull request

This pull request (PR) adds draft release notes for the 2.2.10 patch to the `develop` branch for publication to the beta environment.

### CODE REVIEW

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 1]
```diff
@@ -0,0 +1,794 @@
...
+## Highlights
+
+Look for the following highlights in this release:
+
+### Platform upgrades
+
+The following upgrades to core platform components boost platform security and support PCI compliance:
+
+* Magento 2.2.10 now supports PHP 7.2.x (tested with 7.2.21). <!--- MC-16174 -->
+* Magento 2.2.10 does not support PHP 7.0.x.  <!--- MC-18521 -->
```
Comment (meker12) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]: A nit, but removal of PHP 7.0.x doesn't sound like an upgrade

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 2]
```diff
@@ -0,0 +1,794 @@
...
+This release includes the following security enhancements:
+
+* PSD2 compliance for core payment methods  
+* Fixes for 75 critical security issues
+* Significant platform-security enhancements that boost XSS (cross-site scripting) protection against future exploits. This effort is the culmination of several months of concentrated effort on Magento's part to reduce our backlog of security enhancements.
+
+#### Core payment methods integrations are now compliant with PSD2 regulations
+
+The European Union recently revised the Payment Services Directive (PSD) regulation with an updated version–PSD2. This revised regulation will go into effect on September 14, 2019 or shortly thereafter, and will significantly affect  most payment processing involving credit cards or bank transfers.  See the Magento Forum DevBlog post [3D Secure 2.0 changes](https://community.magento.com/t5/Magento-DevBlog/3D-Secure-2-0-changes/ba-p/136460) for more information on Magento Payment Provider Recommendations and a wealth of links to PSD2 regulation discussions.
+
```
Comment (meker12) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]: General note -- Check for double-spacing between words.

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 3]
```diff
@@ -0,0 +1,794 @@
...
+This release includes the following security enhancements:
+
+* PSD2 compliance for core payment methods  
+* Fixes for 75 critical security issues
+* Significant platform-security enhancements that boost XSS (cross-site scripting) protection against future exploits. This effort is the culmination of several months of concentrated effort on Magento's part to reduce our backlog of security enhancements.
+
+#### Core payment methods integrations are now compliant with PSD2 regulations
+
+The European Union recently revised the Payment Services Directive (PSD) regulation with an updated version–PSD2. This revised regulation will go into effect on September 14, 2019 or shortly thereafter, and will significantly affect  most payment processing involving credit cards or bank transfers.  See the Magento Forum DevBlog post [3D Secure 2.0 changes](https://community.magento.com/t5/Magento-DevBlog/3D-Secure-2-0-changes/ba-p/136460) for more information on Magento Payment Provider Recommendations and a wealth of links to PSD2 regulation discussions.
```
Comment (meker12) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]:
```suggestion
The European Union recently revised the Payment Services Directive (PSD) regulation with an updated version–PSD2. This revised regulation will go into effect on September 14, 2019 or shortly thereafter, and will significantly affect most payment processing involving credit cards or bank transfers.  See the Magento Forum DevBlog post [3D Secure 2.0 changes](https://community.magento.com/t5/Magento-DevBlog/3D-Secure-2-0-changes/ba-p/136460) for more information on Magento Payment Provider Recommendations and a wealth of links to PSD2 regulation discussions.
```

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 4]
```diff
@@ -0,0 +1,794 @@
...
+<!-- ENGCOM-5112 -->
+* Magento no longer caches absolute file paths in  the validator factory  (`Magento\Framework\Validator\Factory::_initializeConfigList`). Previously, caching absolute file paths resulted in problems during transactions when a customer, a `customer_address`, or quote for a registered customer was saved. *Fix submitted by David Führ in pull request [22805](https://github.com/magento/magento2/pull/22805)*. [GitHub-21842](https://github.com/magento/magento2/issues/21842)
+
+<!-- ENGCOM-5217 -->
+* The description of the `setStoreId` function has been amended to more clearly explain how the function helps load CMS pages. *Fix submitted by Prakash Prajapati in pull request [23149](https://github.com/magento/magento2/pull/23149)*. [GitHub-22767](https://github.com/magento/magento2/issues/22767)
+
+### Media Storage
+
+<!-- MAGETWO-93497 -->
+* Media directives now work as expected when an administrator uses a custom domain  (`admin/url/custom`) instead the default domain. Previously,  the WYSIWYG `image-selector` returned malformed media directives, which resulted in broken links to images. [GitHub-16427](https://github.com/magento/magento2/issues/16427)
```
Comment (keharper) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]:
```suggestion
* Media directives now work as expected when an administrator uses a custom domain  (`admin/url/custom`) instead the default domain. Previously, the WYSIWYG `image-selector` returned malformed media directives, which resulted in broken links to images. [GitHub-16427](https://github.com/magento/magento2/issues/16427)
```

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 5]
```diff
@@ -0,0 +1,794 @@
...
+* The Admin Product Edit page and Customers page now load without JavaScript errors. [GitHub-5967](https://github.com/magento/magento2/issues/5967)
+
+<!-- MAGETWO-99722 -->
+* Magento now displays the correct currency in the Admin **Catalog** > **Products**  list  in deployments where websites are assigned different currencies. Previously, the products on the Admin Category page displayed the base currency even when **Product Price Scope** was set to **Per Website** and the website was assigned a different currency.
+
+<!-- MAGETWO-73840 -->
+* Videos in product descriptions now appear as they do in the Admin WYSIWYG editor. Previously, videos in the storefront product descriptions had the incorrect height.
+
+<!-- MC-17021 -->
+* We’ve refined how Magento validates partial permissionsDesign edit permissions for categories, products, and CMS pages are now validated for each endpoint (web API and other) outside of the related model-layer classes. The web API now returns an error when design-related fields are being overridden. Previously, this behavior was ignored.
```
Comment (meker12) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]:
```suggestion
* We’ve refined how Magento validates partial permissions. Design edit permissions for categories, products, and CMS pages are now validated for each endpoint (web API and other) outside of the related model-layer classes. The web API now returns an error when design-related fields are being overridden. Previously, this behavior was ignored.
```

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 6]
```diff
@@ -0,0 +1,794 @@
...
+<!-- MAGETWO-17930 -->
+* You can now edit a customer address from the Admin (**Admin** > **Customer** > **Address** > **Edit Address**) when the customer address attribute is of type `file` or `image`. Previously, Magento did not display the Edit Address form when you clicked on **Edit Address**.
+
+<!-- MAGETWO-99471 -->
+* Custom customer address attribute values are populated as expected when an administrator changes a customer address during order creation from the Admin. Previously, the custom attribute drop-down was empty.
+
+### Customer segment
+
+<!-- MAGETWO-99571 -->
+* You can now create an order from the Admin when  you have a customer segment for customers with 0 or more orders. Previously, if you had a customer segment for customers with 0 or more orders, an SQL error occurred when you tried to create an order in the Admin.
```
Comment (keharper) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]:
```suggestion
* You can now create an order from the Admin when you have a customer segment for customers with 0 or more orders. Previously, if you had a customer segment for customers with 0 or more orders, an SQL error occurred when you tried to create an order in the Admin.
```

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 7]
```diff
@@ -0,0 +1,794 @@
...
+
+### Catalog rule
+
+<!-- MC-18038 -->
+* The CatalogRule module now handles discrepancies between the Magento timezone offset and the system timezone offset (which is in UTC). Previously, when the Magento timezone offset was greater than the system timezone offset, the active ranges set for special prices were inaccurate. This is a consequence of how  catalog price rules special prices are stored and updated. (Catalog price rules special prices are stored in the `catalogrule_product_price` table. This table’s daily update is triggered by the `catalogrule_apply_all` cron job, which is scheduled at 01:00 every day. Cron schedule times are scheduled in Magento timezone.)
+
+<!-- MC-17459 -->
+* Coupon expiration dates and times now match the `end_date` value set in the staging update. Previously, coupon expiration dates could differ from the expiration date set by the sales rule.
+
+<!-- MC-17459 -->
```
Comment (keharper) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]: This is a duplicate.

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 8]
```diff
@@ -0,0 +1,794 @@
...
+
+<!-- MAGETWO-97974 -->
+* You can now update product content descriptions on the store-view level when WYSIWYG is disabled.
+
+### Catalog rule
+
+<!-- MC-18038 -->
+* The CatalogRule module now handles discrepancies between the Magento timezone offset and the system timezone offset (which is in UTC). Previously, when the Magento timezone offset was greater than the system timezone offset, the active ranges set for special prices were inaccurate. This is a consequence of how  catalog price rules special prices are stored and updated. (Catalog price rules special prices are stored in the `catalogrule_product_price` table. This table’s daily update is triggered by the `catalogrule_apply_all` cron job, which is scheduled at 01:00 every day. Cron schedule times are scheduled in Magento timezone.)
```
Comment (meker12) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]: Scheduled at 1:00 a.m. or 1:00 p.m.?

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 9]
```diff
@@ -0,0 +1,794 @@
...
+<!-- MC-17883 -->
+* The Review & Payment section of the One Page checkout no longer displays custom customer attribute code when a guest checks out.
+
+<!-- MAGETWO-98462 -->
+* You can now add product quantities that require four digits to the shopping cart. Previously, Magento could not add four-digit product quantities to the cart.
+
+<!-- ENGCOM-5389 -->
+* The minicart loader is now visible when you add a product to the minicart. *Fix submitted by Prakash Prajapati in pull request [23536](https://github.com/magento/magento2/pull/23536)*. [GitHub-23377](https://github.com/magento/magento2/issues/23377)
+
+<!-- ENGCOM-5400 -->
```
Comment (keharper) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]: Missing a description here

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 10]
```diff
@@ -0,0 +1,794 @@
...
+* Magento now recalculates cart subtotals as expected when one of the ordered products that belongs to a shared catalog  is disabled from the Admin. Previously, when you reloaded the cart after one of its products had been disabled, Magento reloaded the cart page with this exception: `Exception #0 (Magento\Framework\Exception\NoSuchEntityException): The product that was requested doesn't exist. Verify the product and try again`.
+
+<!-- MAGETWO-98700 -->
+* Guests can now access available product options for a configurable product when one of its simple product options is out-of-stock but the configurable product is listed as in-stock in the shared catalog. Previously, under these circumstances, the options drop-down menu for the configurable product was empty, which prevented guests from ordering available options.
+
+<!-- MC-15402 -->
+* Export files now include all columns (including those not visible in the Company list) and their data. Previously, the `State/Province` columns of the exported CSV file were empty.
+
+<!-- MC-17882 -->
+* Request a Quote functionality now works as expected on browsers that are running Internet Explorer 11.x.
```
Comment (keharper) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]:
```suggestion
* Request a Quote functionality now works as expected on Internet Explorer 11.x.
```

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 11]
```diff
@@ -0,0 +1,794 @@
...
+
+### Website restriction
+
+<!-- MAGETWO-93052 -->
+* Administrators with appropriate permissions can now create a new customer account on the Admin when the **Website Restriction** setting is enabled. Previously, Magento threw this exception: `Can not register new customer due to restrictions are enabled`.
+
+### Wish List
+
+<!-- MAGETWO-96873 -->
+* The behavior of the Catalog page Requisition list menu has been corrected.
```
Comment (meker12) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]: This seems a little vague.

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 12]
```diff
@@ -0,0 +1,794 @@
...
+### URL rewrite
+
+<!-- MAGETWO-72788 -->
+* You can now export newsletter subscribers from the Admin. Previously, Magento displayed this error when you selected a subscriber name and clicked **Export**: `error: URI too long`
+
+<!-- MAGETWO-96662 -->
+* Magento no longer removes the query string from URLs when the query string is preceded by a slash. Previously, when a customer opened a URL that contained a trailing slash and query string (for example, `http://magento.host.com/sample-url-key/?cupcakes`), Magento redirected the user to a URL that omitted the slash (`http://magento.host.com/sample-url-key`).
+
+<!-- MAGETWO-99925 -->
+* Products are successfully updated through import of an CSV file in **Add/Update** mode. Previously,  the import process failed, and Magento displayed this error: `The value specified in the URL Key field would generate a URL that already exists`.
```
Comment (meker12) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]:
```suggestion
* Products are successfully updated through import of a CSV file in **Add/Update** mode. Previously,  the import process failed, and Magento displayed this error: `The value specified in the URL Key field would generate a URL that already exists`.
```

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 13]
```diff
@@ -0,0 +1,794 @@
...
+<!-- ENGCOM-5394 -->
+* The behavior of the mobile menu JavaScript now triggers at the same breakpoint as the mobile menu styles. *Fix submitted by bobemoe in pull request [23547](https://github.com/magento/magento2/pull/23547)*. [GitHub-8298](https://github.com/magento/magento2/issues/8298)
+
+<!-- ENGCOM-5396 -->
+* The **Refund** button on the credit memo page now remains active after a merchant enters a value in the Refund Totals section. *Fix submitted by Prakash Prajapati in pull request [23566](https://github.com/magento/magento2/pull/23566)*. [GitHub-23285](https://github.com/magento/magento2/issues/23285)
+  
+<!-- ENGCOM-5336 -->
+* Magento now displays the cursor to the right of the search keyword box as expected after multiple clicks on the search field in mobile view. *Fix submitted by Prakash Prajapati in pull request [23352](https://github.com/magento/magento2/pull/23352)*. [GitHub-22736](https://github.com/magento/magento2/issues/22736)
+  
+<!-- ENGCOM-5401 -->
```
Comment (meker12) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]: Duplicated on line 694.

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md (hunk 2 above)
Comment (jeff-matthews) [reply guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]: Did a global search and replace in both files

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 14]
```diff
@@ -0,0 +1,794 @@
...
+
+This release contains the following major PSD-related changes:
+
+* The **Braintree payment method now complies with PSD2 regulations**. Its core integration API has been upgraded to the latest JavaScript SDK v3 API, which is a requirement for supporting native Braintree 3D Secure 2.0 adoption. Braintree transactions are now also verified by using the native Braintree 3D Secure 2.0 service. <!--- MAGETWO-99607 MC 17628 -->
+
+* Authorize.net now provides the ability, through the `cardholderAuthentication` request field, to make 3D Secure verification through third-party services such as CardinalCommerce. Starting with this release, **Authorize.net Accept.js integration will support 3DS 2.0 through CardinalCommerce**. <!--- MAGETWO-99737 -->
+
+<!--- MC-18237 -->
+* The Cybersource and eWay payment modules have been deprecated in this release to comply with PSD2 SCA regulation, which takes effect on September 14, 2019, or shortly thereafter. Use the official Marketplace extensions for these features instead.
```
[the remark this answers is not in the document]
Comment (jeff-matthews) [reply guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]: Updated verb tense

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 15]
```diff
@@ -0,0 +1,794 @@
...
+
+### Reward
+
+<!-- MAGETWO-99388 -->
+* Online refunds now work as expected when the **Refund Reward Points Automatically** configuration setting is enabled. Previously, the **Refund** button was disabled under these conditions.
+
+### RMA
+
+<!-- MAGETWO-73749 -->
+* Clicking **Show Packages** on a Returns page (**My Account** > **My Returns**  > **Return**) now opens a new page about the selected package.Previously, clicking on this link resulted in a 404 error page.
```
Comment (meker12) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]:
```suggestion
* Clicking **Show Packages** on a Returns page (**My Account** > **My Returns**  > **Return**) now opens a new page about the selected package. Previously, clicking on this link resulted in a 404 error page.
```

**Comment by jeff-matthews** [comment]
running tests

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md [hunk 16]
```diff
@@ -0,0 +1,789 @@
...
+<!-- ENGCOM-5392 -->
+* All emails are now sent with correct MIME encoding. *Fix submitted by gwharton in pull request [23537](https://github.com/magento/magento2/pull/23537)*. [GitHub-22103](https://github.com/magento/magento2/issues/22103), [GitHub-23199](https://github.com/magento/magento2/issues/23199)
+
+<!-- ENGCOM-5437 -->
+* Email created using a CSS-heavy template is now sent successfully. Previously, these emails were rejected by the server with this message: `Message too big`. *Fix submitted by gwharton in pull request [23650](https://github.com/magento/magento2/pull/23650)*. [GitHub-23643](https://github.com/magento/magento2/issues/23643)
+
+### Frameworks
+
+<!-- MAGETWO-99872 -->
+* The `equalArrays` function in `lib/web/mage/utils/compare.js` file has been refactored to remove quadratic complexity. Previously, this feature significantly slowed Admin operations that were performed on large number of products (for example, adding a product to category by SKU).
```
Comment (meker12) [inline guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]:
```suggestion
* The `equalArrays` function in `lib/web/mage/utils/compare.js` file has been refactored to remove quadratic complexity. Previously, this feature significantly slowed Admin operations that were performed on a large number of products (for example, adding a product to category by SKU).
```

File: guides/v2.2/release-notes/release-notes-2-2-10-commerce.md (hunk 13 above)
Comment (jeff-matthews) [reply guides/v2.2/release-notes/release-notes-2-2-10-commerce.md]: Removed this dupe

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**